pkg/operator/operator.go

```diff
@@ -116,15 +117,13 @@ func New(config operatorconfig.Config, kubeConfig *rest.Config) (*Operator, erro
 		return nil, fmt.Errorf("failed to create dns controller: %v", err)
 	}
 
-	// Set up the canary controller when the config.CanaryImage is not empty
-	if len(config.CanaryImage) != 0 {
```

Comment:
Is there any particular reason to remove this check?  Well, I guess I don't see a need to keep it.

Reply:
We originally had this check in place so that the canary could be disabled when running the operator locally as per Dane's request. See https://github.com/openshift/cluster-ingress-operator/blob/master/hack/run-local.sh
(although I agree its not strictly necessary)

Reply:
That's a good reason to keep it, I'll add it back.